tokio/src/net/udp/socket.rs

```diff
@@ -303,6 +340,59 @@ impl UdpSocket {
             .await
     }
 
+    /// Attempts to receive a single datagram message on the socket from the remote
+    /// address to which it is `connect`ed.
+    ///
+    /// The [`connect`] method will connect this socket to a remote address. The future
+    /// will resolve to an error if the socket is not connected..
+    ///
+    /// Note that on multiple calls to a `poll_*` method in the recv direction, only the
+    /// `Waker` from the `Context` passed to the most recent call will be scheduled to
+    /// receive a wakeup.
+    ///
+    /// # Return value
+    ///
+    /// The function returns:
+    ///
+    /// * `Poll::Pending` if the socket is not ready to read
+    /// * `Poll::Ready(Ok(n))` `n` is the number of bytes read.
+    /// * `Poll::Ready(Err(e))` if an error is encountered.
+    ///
+    /// # Errors
+    ///
+    /// This function may encounter any standard I/O error except `WouldBlock`.
+    ///
+    /// [`connect`]: method@Self::connect
+    pub fn poll_recv(
+        &self,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<usize>> {
```

Comment:
Since it uses `ReadBuf`, it should return `Result<()>` for consistency with `AsyncReadExt::read`.

Reply:
FYI, `poll_peek` from `TcpStream` is ` -> Poll<io::Result<usize>>`

Reply:
I was going to reply saying `poll_peek` on `TcpStream` returns the size, but I just noticed it also does not use `ReadBuf`. That's something that's been overlooked, yes? I can add a quick fix there also.

Reply:
Yes, `TcpStream::poll_peek` was overlooked. See #2987.